### DISCUSSION TITLE
Conditional Parsing values not being added to size

### DISCUSSION THREAD

**Discussion by AidanWelch** [WerWolv/ImHex#1503] [Get Help] [answered]
Hi,

I have an array of values, of a type where they have conditional members.  Unfortunately, the size seems to be computed only from the static properties.  Is there a way around this?

![Screenshot 2024-01-19 140758](https://github.com/WerWolv/ImHex/assets/35280359/670b63a9-a889-4311-8cda-205a36432012)

In the example above `collection` is a conditional property.

**Comment by AidanWelch** [comment]
I think it may be because its a conditional union

**Comment by AidanWelch** [comment, marked as answer]
Yeah, condition union was the issue.  I was able to resolve it by making it a struct then offsetting my other structs with `$ -= 0x02;`

**Comment by paxcut** [comment]
From what you posted the unions are the members, but the collection is a structure. the size of the union is the size of the largest element in the union. If the collection is part of the first union, then the size of the union should be 0xD and not 2. I have tested that with a simple example and it works.
the code
```cpp
struct test {
   u32 array[5];
};

union test2 {
    u8 a;
    u8 b;
    test t;
};

test2 t2 @0;
```
and the result
![image](https://github.com/WerWolv/ImHex/assets/53811119/b805a8a3-20e5-4fe4-855c-a2d8d8ef2350)

**Reply by AidanWelch** [reply]
Except the size is not correct for conditional unions:

```c
struct test {
   u32 array[5];
};

union test2 {
    u8 a;
    u8 b;
    if (a == b){
        test t;
    }
};

test2 t2 @0;
```

![Screenshot 2024-01-19 152515](https://github.com/WerWolv/ImHex/assets/35280359/130dec63-58aa-4f5b-8359-1ab4a1d96da3)

**Comment by paxcut** [comment]
I see. You marked your response as a solution but it just a work around and the bug needs to be fixed because unions should also calculate their sizes correctly. In order to narrow down the possible places where the bug is located you may want to mention unions in the title.

**Reply by AidanWelch** [reply]
This isn't a bug report though, I wasn't sure if I was misunderstanding what the intended effect is.  I will open an issue to report it as an actual bug now that I know it is one.

**Reply by AidanWelch** [reply]
I opened #1504

**Reply by paxcut** [reply]
I saw that and it look very good. straight to the point and even provided simple code. I appreciate you taking the time to do this.

**Reply by AidanWelch** [reply]
Of course!  I appreciate the effort to improve and maintain a tool that's been very helpful to me!
